# Regular computer simulator

## How computer works?

Computer takes input, processes it and gives output:

In [32]:
# input
a = 0x00000001
b = 0x00000111

# processing
c = a + b

# output
print(f"output: {c}")

output: 274


Output should be `274` in decimal or `0x00000112` in hex.

## How computer stores data?

Computer stores data in memory. Memory is a big array of bytes. Each byte has its own address. For example, address of first byte is `0x00000000`, address of second byte is `0x00000001` and so on. Each byte can store number from `0x00` to `0xFF` in hex or from `0` to `255` in decimal. For example, `0x00` is `0` in decimal, `0x01` is `1` in decimal, `0x0A` is `10` in decimal, `0xFF` is `255` in decimal.

In [24]:
first_byte = 0x00
second_byte = 0x01
third_byte = 0x02
# ...
tenth_byte = 0x09

print(f"first byte: {first_byte}")
print(f"second byte: {second_byte}")
print(f"third byte: {third_byte}")
print(f"tenth byte: {tenth_byte}")
print(tenth_byte)

first byte: 0
second byte: 1
third byte: 2
tenth byte: 9


TypeError: expected 1 argument, got 0